chemprop/cli/predict.py

```diff
@@ -149,9 +149,16 @@ def add_predict_args(parser: ArgumentParser) -> ArgumentParser:
 
 
 def process_predict_args(args: Namespace) -> Namespace:
-    args.output = args.output or args.test_path.parent / (args.test_path.stem + "_preds.csv")
+    if args.test_path.suffix not in [".csv"]:
+        raise ArgumentError(
+            argument=None, message=f"Input data must be a CSV file. Got {str(args.test_path)}"
+        )
+    if args.output is None:
+        args.output = args.test_path.parent / (args.test_path.stem + "_preds.csv")
     if args.output.suffix not in [".csv", ".pkl"]:
-        args.output = Path(str(args.output) + ".csv")
+        raise ArgumentError(
+            argument=None, message=f"Output must be a CSV or Pickle file. Got {str(args.output)}"
```

Comment:
f-strings implicitly call `str` on all inputs. `str` in turn defaults to `repr` if the object has no `__str__` method defined.
```suggestion
            argument=None, message=f"Output must be a CSV or Pickle file. Got {args.output}"
```